sphinxcontrib/needs/needs.py

```diff
@@ -514,6 +516,20 @@ def check_configuration(_app: Sphinx, config: Config):
     extra_options = config["needs_extra_options"]
     link_types = [x["option"] for x in config["needs_extra_links"]]
 
+    log = get_logger(__name__)
+    external_filter = getattr(config, "needs_filter_data", {})
+    for extern_filter, value in external_filter.items():
+        # Check if external filter values is really a string
+        if not isinstance(value, str):
+            log.warning(
```

Comment:
I think this should raise an exception as well.
Not having string-values in our data may be a huge problem, as our code does not check this everywhere.